backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/cfg/ElasticsearchIndexManagementStrategyConfiguration.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.cfg;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Strategy for creating/deleting the indexes in Elasticsearch upon Hibernate Search initialization and shut-down.
+ *
+ * @author Gunnar Morling
+ * @author Yoann Rodiere
+ */
+public enum ElasticsearchIndexManagementStrategyConfiguration {
```

Comment:
Maybe we can change this name is something more synthetic, such as `ElasticsearchIndexCreationStrategy`